### ISSUE TITLE
iRecord download issue WCBS-BBS 

### ISSUE THREAD

**Issue by zrandle** [BiologicalRecordsCentre/UKBMS-online#293]
A county recorder has raised this issue
 
I've just downloaded the section level data for WCBS-BBS from the UKBMS website and found a problem with the section gridrefs. They have all been provided as "easting, northing" e.g. "490171, 226313" and not a normal OS gridref. Last year they were all presented as normal gridrefs. I could spend time extracting and rearranging bits of text to form into OS gridrefs, but I shouldn't have to!
 
By comparison the WCBS-BC data section gridrefs are fine, e.g. SP942314.
 
Please can you investigate why this has happened?

**Comment by DavidRoy** [comment]
@birenrathod - would you be able to look at this please

**Comment by BirenRathod** [comment]
@zrandle is it ok to pass me the url of that page from ukbms website?

**Comment by zrandle** [comment]
@RachaelConway do you know what the URL would be for this page?

**Comment by RachaelConway** [comment]
Url for the page is https://ukbms.org/annual-summary

Need to select field options to WCBS BBS as below
![image](https://user-images.githubusercontent.com/88440885/201965165-b78316ed-efde-4018-be14-ed4cd6ce91e3.png)


Then Downloads  - https://ukbms.org/annual-summary?userID=branch&report-location_type_id=5196

Section level - download CSV
![image](https://user-images.githubusercontent.com/88440885/201964846-cc240840-68ee-4b27-afbf-458ffcfeb850.png)

**Comment by BirenRathod** [comment]
Thanks @RachaelConway. When I downloaded same it has got gridref in it. Which user did you log-in with? See below screenshot.
![image](https://user-images.githubusercontent.com/5941255/201978388-f24d27d1-a8e0-4ea7-b7be-cbdc2bad084f.png)

**Comment by zrandle** [comment]
@BirenRathod @RachaelConway the recorder was Keith Balmer and this is the URL he sent me https://ukbms.org/annual-summary?report-location_type_id=5195&userID=branch

**Comment by BirenRathod** [comment]
Thanks, @zrandle. I'm investigating into it.

**Comment by RachaelConway** [comment]
@BirenRathod Just for info, I was using my own account and recreated the issue using my branch data

**Comment by BirenRathod** [comment]
@RachaelConway I found that data stored for WCBS-BBS is using different geometry than WCBS-BC for some reason. Even though settings are same for both.
@johnvanbreda any idea?

**Comment by DavidRoy** [comment]
@burkmarr I think you might have loaded these layers?  Any idea on resolving the issue?

**Comment by burkmarr** [comment]
@DavidRoy - I'll look into it.

**Comment by burkmarr** [comment]
We added all the section geometries for the  WCBS-BBS squares back in March (see https://github.com/BiologicalRecordsCentre/UKBMS-online/issues/255). I used the location upload facility of the Warehouse which automatically creates the centroid sref (which corresponds to the section grid ref) as an easting/northing pair. (There is no option to create or upload a grid reference for the centroid sref.) 

I think I will have to use SQL to update the centroid_sref from easting, northing to GR from a temporary table. There is a corresponding field - centroid_geom - which contains the geometry of the centroid which would also need updating to the geometry corresponding to the square of the centroid GR. I think that I can create the temporary table from a shapefile using the approach mentioned by @johnvanbreda here (https://github.com/BiologicalRecordsCentre/UKBMS-online/issues/255#issuecomment-1062862692). 

@zrandle would you like the section grid refs recorded as 6 figure or 8 figure grid references? (I see that WCS-BC uses mostly 6 figure, but some use 8 figure.)

**Comment by burkmarr** [comment]
@johnvanbreda @BirenRathod @JimBacon -  can any of you help me with this question: in the locations table, what coordinate system is the centroid_geometry stored in. Does it depend on the centroid_sref_system or is always constant (e.g. wgs84 - 4326)?

**Comment by BirenRathod** [comment]
@burkmarr As long as I understand this, centroid_sref_system is for storing constent which geometry stored in centroid_geometry. it is not depend on it. so it is possible to stored different geometry without storing constent in centroid_sref_system for example.

**Comment by JimBacon** [comment]
Hi @burkmarr,

I think all geometries are stored in the google web mercator projection. It is configurable and set in the warehouse application/config/sref_notations.php file with key, 'internal_srid'. The centroid_sref_system will just refer to the centroid_sref.

**Comment by RachaelConway** [comment]
@IanMiddlebrook Hi Ian, Zoe has already opened the section level GRID ref issue here

**Comment by IanMiddlebrook** [comment]
@burkmarr  
_would you like the section grid refs recorded as 6 figure or 8 figure grid references? (I see that WCS-BC uses mostly 6 figure, but some use 8 figure_

If it's possible to distinguish between squares where section routes are mapped out, and those that are not ... then 8 figure section grid refs for mapped sections, and 6 figure grid refs for others.

There was a previous complaint from recorders about 8 figure grid refs set up as a diagonal line from SW to NE corner of the square, when these did not represent an actual route. So this should be avoided.

**Comment by burkmarr** [comment]
Thanks @JimBacon and @IanMiddlebrook.

**Comment by JimBacon** [comment]
Just to confirm, the geometry columns are created in `modules/indicia_setup/db/indicia_tables.sql` with the spatial reference system hard-coded at 900913, the initial number for the Google Web Mercator projection.

**Comment by burkmarr** [comment]
@zrandle - I think that this is complete now. I have upated all the WCS-BBS transects that had easting/northing instead of GR to use 8 fig GR - that was around 60000 transect sections. @IanMiddlebrook - all of these had linear geometry so I used 8 figure GR for all of them. I followed the procedure described about by @RachaelConway to test and all appeared to be well. Let me know if there are any problems. Thanks @JimBacon for the helpful discussion this morning.
